lib/server.js

```diff
@@ -391,28 +397,34 @@ Server.prototype.onWebSocket = function (req, socket) {
   if (id) {
     var client = this.clients[id];
     if (!client) {
-      debug('upgrade attempt for closed client');
+      debugError('upgrade attempt for closed client: %s', id);
       socket.close();
     } else if (client.upgrading) {
-      debug('transport has already been trying to upgrade');
+      debugError('transport has already been trying to upgrade for client [%s]', id);
       socket.close();
     } else if (client.upgraded) {
-      debug('transport had already been upgraded');
+      debugError('transport had already been upgraded for client [%s]', id);
       socket.close();
     } else {
       debug('upgrading existing transport');
 
       // transport error handling takes over
       socket.removeListener('error', onUpgradeError);
 
-      var transport = new transports[req._query.transport](req);
-      if (req._query && req._query.b64) {
-        transport.supportsBinary = false;
-      } else {
-        transport.supportsBinary = true;
+      try {
```

Comment:
I was not sure but in https://github.com/socketio/engine.io/blob/master/lib/server.js#L299 
Same statement wrapped in `try..catch` so I did same.